server/xpub-controller/helpers/files.js

```diff
@@ -0,0 +1,154 @@
+const logger = require('@pubsweet/logger')
+const FileModel = require('@elifesciences/xpub-model').File
+
+class FilesHelper {
+  constructor(config, scienceBeamApi) {
+    this.config = config
+    this.scienceBeamApi = scienceBeamApi
+  }
+
+  validateFileSize(fileSize) {
+    if (fileSize > this.config.get('fileUpload.maxSizeMB') * 1e6) {
+      throw new Error(
+        `File size shouldn't exceed ${this.config.get(
+          'fileUpload.maxSizeMB',
+        )}MB`,
+      )
+    }
+  }
+
+  static async generateFileEntity(file, manuscriptId) {
+    const { stream, filename, mimetype: mimeType } = await file
+
+    const fileEntity = await new FileModel({
+      manuscriptId,
+      url: `manuscripts/${manuscriptId}`,
+      filename,
+      type: 'MANUSCRIPT_SOURCE_PENDING',
+      mimeType,
+    }).save()
+    return {
+      stream,
+      fileEntity,
+    }
+  }
+
+  static startFileProgress(
+    pubsub,
+    ON_UPLOAD_PROGRESS,
+    startedTime,
+    predictedTime,
+    manuscriptId,
+  ) {
+    return setInterval(() => {
```

Comment:
To make this function easier to test, lets remove the `setInterval` at this level and re-name this something like `publishPredictedProgress` ? The `setInterval` could then be called at the controller level.